### DISCUSSION TITLE
Does Mastondon have an API for interfacing with IFTTT apps?

### DISCUSSION THREAD

**Discussion by steevithak** [mastodon/mastodon#21368] [Q&A] [answered]
I'm a new Mastodon user trying to migrate from Twitter but I'm having trouble connecting my other social media accounts to Mastodon. Typically on Twitter, I would post tweets directly as a well as generate automated tweets from my Instagram, Wordpress Blog, Flickr or and other stuff. It was a simple matter of browsing IFTTT for things like an IG-to-Twitter app to connect them. But there seem to be no such apps for Mastodon (on IFTTT or elsewhere that I've found).

I'm curious if this is because the Mastodon code base has no API to allow the creation of such apps? Or is the API there but no one has used it to build apps to get data in and out of Mastodon yet? If the latter, could someone point me to the API docs so maybe I could try creating some apps?

**Comment by nightpool** [comment, marked as answer]
Unfortunately, my understanding is that every IFTTT integration is manual and controlled entirely by the proprietary decisions of the IFTTT team, and requires substantial effort to implement IFTTT's custom logic. If the IFTTT would be interested in adding ActivityPub/Mastodon support, we'd be happy to talk to them about the APIs they can use to achieve that, but unfortunately it doesn't seem like they're interested. 

You can view the docs for Mastodon's APIs at https://docs.joinmastodon.org/client/intro/ and there are already a huge number of third party apps, client libraries, etc., just nothing for IFTTT.

**Reply by steevithak** [reply]
Any suggestions on where to look for the "huge number of third party apps"? Is there a community repository of Mastodon apps or a list of links where I could start? My main interest to start with is in finding a simple Instagram-to-Mastodon app so my IG posts will propagate to my Mastodon account. Maybe there's an open source equivalent to IFTTT out there for cross-API apps that I haven't found yet?

**Reply by nightpool** [reply]
It looks like Moa (a popular cross-poster) used to support instagram-to-mastodon crossposting, but Facebook changed the API and broke it: https://github.com/FedStoa/moa/issues/156. As to where/how to find third-party apps, I can only really recommend Google, we used to have a list of apps but it was too long and too frequently changing.
